apps/entity-service/src/entities/medias.controller.ts

```diff
@@ -0,0 +1,63 @@
+import {
+  BadRequestException,
+  Controller,
+  Delete,
+  NotFoundException,
+  Param,
+  Query,
+  UnauthorizedException
+} from "@nestjs/common";
+import { ExceptionResponse, JsonApiResponse } from "@terramatch-microservices/common/decorators";
+import { MediaDto } from "../entities/dto/media.dto";
+import { ApiOperation } from "@nestjs/swagger";
+import { MediaService } from "@terramatch-microservices/common/media/media.service";
+import { PolicyService } from "@terramatch-microservices/common/policies/policy.service";
+import { buildDeletedResponse } from "@terramatch-microservices/common/util/json-api-builder";
+import { TMLogger } from "@terramatch-microservices/common/util/tm-logger";
+
+@Controller("entities/v3/medias")
+export class MediasController {
+  private logger = new TMLogger(MediasController.name);
+
+  constructor(private readonly mediaService: MediaService, private readonly policyService: PolicyService) {}
+
+  @Delete("bulkDelete")
+  @ApiOperation({
+    operationId: "mediaBulkDelete",
+    summary: "Delete multiple media"
+  })
+  @JsonApiResponse({ data: MediaDto })
+  @ExceptionResponse(UnauthorizedException, { description: "Authentication failed." })
+  @ExceptionResponse(BadRequestException, { description: "Invalid request." })
+  @ExceptionResponse(NotFoundException, { description: "Resource not found." })
+  async mediaBulkDelete(@Query() { uuids }: { uuids: string[] }) {
+    this.logger.debug("Bulk deleting media with uuids: " + uuids);
+    const permissions = await this.policyService.getPermissions();
+    if (!permissions.includes("media-manage")) {
+      throw new UnauthorizedException();
+    }
```

Comment:
This should be handled in the policy. Likely by getting the media owner processor and checking for `deleteFiles` like we do for `uploadFiles`. Controllers should never depend directly on permissions - it's the job of the policies to convert permissions into permissible actions.

This has an additional problem as well: it's allowing the greenhouse account to delete any file it can find the UUID for.

Update: I see now that down below you're checking the policy. That's good, but it makes this check weird and superfluous. Given that it is limiting permissions to the owner of the file, I think just removing this `media-manage` check should be enough for this endpoint.

Reply:
Ok, I am leaving this permission check on the policy and I updated the policy check to each subject.
Here is the [reference](https://github.com/wri/wri-terramatch-api/blob/be036d5e36a40420201cd1bfe10c67276fa7b58b/app/Http/Controllers/V2/MediaController.php#L46)